src/datasets/arrow_dataset.py

```diff
@@ -5485,6 +5490,8 @@ def push_to_hub(
             }
         else:
             metadata_config_to_dump = {"data_files": [{"split": split, "path": f"{data_dir}/{split}-*"}]}
+        if set_default_config:
+            metadata_config_to_dump["default"] = True
```

Comment:
You might also have to remove the default field for all the already existing configurations (we should end up with only one configuration with default: True)

Reply:
I was thinking of that...

Indeed this is already the case: currently, multiple configuration can be defined in the YAML as default, and we are returning the first one.

Reply:
It is already the case that we cannot have a configuration named "default" as non-default...